### PR TITLE
CPM-863: Be able to reorder conditions

### PR DESCRIPTION
**Pull request by antoinearrive**
**Description (for Contributor and Core Developer)**
image

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx [hunk 1]
```diff
@@ -60,28 +66,30 @@ const SelectionTab: React.FC<SelectionTabProps> = ({target, conditions, onChange
...
-              {conditionsWithId.map(conditionWithId => (
-                <>
-                  {conditionWithId.type === CONDITION_NAMES.ENABLED && (
-                    <EnabledLine condition={conditionWithId} key={conditionWithId.id} onChange={handleChange} />
-                  )}
-                </>
-              ))}
-            </>
+            <Table.Row>
+              <Styled.NotDraggableCell></Styled.NotDraggableCell>
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:70]:
```suggestion
              <Styled.NotDraggableCell/>
```
There is no need to have both opening and closing tags when there's no children

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx [hunk 2]
```diff
@@ -60,28 +66,30 @@ const SelectionTab: React.FC<SelectionTabProps> = ({target, conditions, onChange
...
+              </Table.Cell>
+            </Table.Row>
           )}
         </Table.Body>
       </Table>
+      <Table isDragAndDroppable={true} onReorder={onReorder}>
+        <Table.Body>
+          {!isLoading &&
+            conditionsWithId.map((row, index) => (
+              <Table.Row key={`${index}`}>
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:87]: Indexes are not recommanded as keys, is there no other data in row that could be used instead? (maybe the row id?)

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx [hunk 3]
```diff
@@ -60,28 +66,30 @@ const SelectionTab: React.FC<SelectionTabProps> = ({target, conditions, onChange
...
+                <EnabledLine key={index} condition={row} onChange={handleChange} />
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:88]: The prop `key` is used by React to know which element needs rerender on a list of items. So it's only needed directly below a `.map` (here you added it correctly to `Table.Row`)
[the doc explains it well ](https://reactjs.org/docs/lists-and-keys.html)

**Review by cemorin — APPROVED** [review]

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx (hunk 2 above)
Comment (pierallard) [reply components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:87]: :+1: Please use Identifiers as the structure table

**Comment by pierallard** [comment]
I suggest you wait for my PR https://github.com/akeneo/pim-community-dev/pull/18735/files#diff-2d36a6b57813a5e83975101c0f9ed2a9168f39c608fa0153a4a276e2a19d8fcbL16
You will have identifiers in it.

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx (hunk 2 above)
Comment (antoinearrive) [reply components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx]: Done

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx [hunk 4]
```diff
@@ -81,6 +87,7 @@ const SelectionTab: React.FC<SelectionTabProps> = ({target, conditions, onChange
     }
 
     closeModal();
+    // eslint-disable-next-line
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:90]: There was en error?

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx [hunk 5]
```diff
@@ -68,6 +71,12 @@ const SelectionTab: React.FC<SelectionTabProps> = ({target, conditions, onChange
     onChange(removeIdentifiers(newConditions));
   };
 
+  const onReorder = (indices: number[]) => {
+    const newConditions = indices.map(i => conditionsWithId[i]).filter(e => !!e);
```
Comment (pierallard) [inline components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:75]: Why filtering ?

**Review by pierallard — APPROVED** [review]

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx (hunk 5 above)
Comment (antoinearrive) [reply components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx]: I just deleted it it was a leftover

**Pull request merged**